fuse_optimizers/CMakeLists.txt

```diff
@@ -6,6 +6,7 @@ set(build_depends
   fuse_constraints
   fuse_core
   fuse_graphs
+  fuse_variables
```

Comment:
This could be on another PR. I simply noticed that `fuse_optimizers` uses `fuse_variables` here: https://github.com/locusrobotics/fuse/blob/devel/fuse_optimizers/src/variable_stamp_index.cpp#L133

It's only part of the implementation though. It's not in the header file.

I can move this to another PR if you want. It got into this commit when I was rebasing all the commits I did during the development of this feature.